### PR TITLE
React test

### PR DESCRIPTION
**Pull request by soesandarwin2201**
## What does this PR do?
This PR is raised to write unit tests for the Math Magicians app, using the Jest and React Testing Library libraries.

General requirements

## Description of the task to be completed
- [x] create the react-test branch 
- [x] install the react-test-renderer
- [x] Set up React Testing Library.
- [x] Write unit tests for the files operate.js and calculate.js using Jest.
- [x] Create unit tests for all React components:
- [x] Use Jest snapshots to test the components.

## How should it be tested
Clone the repository
```
git clone  https://github.com/soesandarwin2201/React-Project
```
Navigate to the folder
```
cd React-Project
```
Checkout the branch
```
git checkout react-test
```
#### Thanks for reviewing my PR :blush:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,6 +49,8 @@
     "eslint-plugin-jsx-a11y": "^6.6.1",
     "eslint-plugin-react": "^7.31.11",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "jest": "^27.5.1",
+    "react-test-renderer": "^18.2.0",
```
Comment (GzimAsani) [inline package.json:53]: - [x] One of your tests is failing, kindly update it 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @soesandarwin2201                              :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the tests, they cover everything :muscle:  :+1: 

 ## Required Changes 
- [ ] You have just a small improvement to make, you are almost there 99% of the job is done, kindly check the comment below :arrow_double_down: 



 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @soesandarwin2201, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (soesandarwin2201) [reply package.json:53]: Thanks for pointing out the right thing . I found the bugs thanks to you and fix that error . Thanks for helping me improve my code.

**Pull request merged**